uPortal-core/src/main/java/org/apereo/portal/portlet/om/PortletLifecycleState.java

```diff
@@ -15,21 +15,34 @@
 package org.apereo.portal.portlet.om;
 
 public enum PortletLifecycleState {
+
     CREATED(0),
     APPROVED(1),
     PUBLISHED(2),
     EXPIRED(3),
     MAINTENANCE(4);
 
-    public static final String MAINTENANCE_MODE_PARAMETER_NAME =
-            "PortletLifecycleState.inMaintenanceMode";
-
     private int order;
 
-    private PortletLifecycleState(int order) {
+    PortletLifecycleState(int order) {
         this.order = order;
     }
 
+    public static PortletLifecycleState forOrderValue(int orderValue) {
+        PortletLifecycleState rslt = null;
+        for (PortletLifecycleState state : PortletLifecycleState.values()) {
+            if (state.getOrder() == orderValue) {
+                rslt = state;
+                break;
+            }
+        }
+        if (rslt == null) {
```

Comment:
Okay, thank makes sense. :+1: 

It still feels a bit round-about to me, the intent is filtering the list, checking if the filtered list is empty, then getting the last element.
With a for loop the order is: filtering and getting the last element the same time, then checking is empty through null check.
Streams can put those operations in an order that gives some more clarity.